src/libs/blueprint/conduit_blueprint_mesh_utils_iterate_elements.hpp

```diff
@@ -567,6 +581,197 @@ traverse_structured(FuncType &&func, const Node &topo)
+        else
+        {
+            // There are no polyhedra.
+            for(index_t ei = 0; ei < nzones; ei++)
+            {
+                const auto size = sizes[ei];
+                const auto offset = offsets[ei];
+                const auto shape = shapes[ei];
+
+                entity e;
+                e.entity_id = ei;
+                e.shape = utils::ShapeType(value2id[shape]);
+                e.element_ids.resize(size);
+                for(index_t j = 0; j < size; j++)
+                {
+                    // Pull out vertex id at ei then cast to index_t
+                    e.element_ids[j] = conn[offset + j];
+                }
+
+                func(e);
+            }
+        }
+    }
+    else
+    {
+        // No offsets so we'll just iterate through the connectivity.
+
+        if(hasPolyhedra && topo.has_path("subelements/connectivity"))
+        {
+            // The elements contain polyhedra.
+            const index_t_accessor se_conn = topo["subelements/connectivity"].as_index_t_accessor();
+            const index_t_accessor se_sizes = topo["subelements/sizes"].as_index_t_accessor();
+            const index_t_accessor se_offsets = topo["subelements/offsets"].as_index_t_accessor();
+
+            index_t offset = 0;
+            for(index_t ei = 0; ei < nzones; ei++)
+            {
+                const auto shape = shapes[ei];
+
+                // Copy basic shape data into the entity.
+                entity e;
+                e.entity_id = ei;
+                e.shape = utils::ShapeType(value2id[shape]);
+
+                // Get size from shape.
+                const auto size = e.shape.indices;
+
+                e.element_ids.resize(size);
+                for(index_t j = 0; j < size; j++)
+                {
+                    e.element_ids[j] = conn[offset + j];
+                }
+
+                // Copy subelement data for polyhedral zones.
+                if(shape == phShape)
+                {
+                    copy_subelements(e, se_conn, se_sizes, se_offsets);
+                }
+
+                func(e);
+
+                offset += size;
+            }
+        }
+        else
+        {
+            // No offsets so we'll just iterate through the connectivity.
+            index_t offset = 0;
+            for(index_t ei = 0; ei < nzones; ei++)
+            {
+                const auto shape = shapes[ei];
+                entity e;
+                e.entity_id = ei;
+                e.shape = utils::ShapeType(value2id[shape]);
+
+                // Get size from shape.
+                const auto size = e.shape.indices;
+
+                e.element_ids.resize(size);
+                for(index_t j = 0; j < size; j++)
+                {
+                    // Pull out vertex id at ei then cast to index_t
+                    e.element_ids[j] = conn[offset + j];
+                }
+
+                func(e);
+
+                offset += size;
+            }
+        }
+    }
+}
```

Comment:
thanks for adding this!

Reply:
Hey @cyrush , I commented out the hdf5 I/O lines in that test since they were mostly for debugging and looking at baseline data. I think this is good to go.

Reply:
Awesome, thanks again for adding this!